File: frontend/AI_TradingAgent/types/trading.ts
export type AssetType = "stock" | "etf" | "option" | "crypto";

export type OrderSide = "buy" | "sell";

export type OrderType = "market" | "limit" | "stop";

export type TimeInForce = "day" | "gtc";

export type MarketStatus =
  | "open"
  | "closed"
  | "pre-market"
  | "after-hours"
  | "unknown";

export type AgentStatus = "active" | "paused" | "evaluating";

export type MarketRegime =
  | "bullish"
  | "bearish"
  | "neutral"
  | "high-volatility"
  | "low-volatility";

export type RiskLevel = "low" | "moderate" | "high" | "critical";

export type ChartTimeframe =
  | "1D"
  | "5D"
  | "1M"
  | "3M"
  | "6M"
  | "1Y";

export interface Quote {
  symbol: string;
  price: number;
  change: number;
  changePercent: number;
  volume: number;
  marketStatus: MarketStatus;
  timestamp: string;
}

export interface ChartPoint {
  timestamp: string;
  price: number;
  volume: number;
  open: number;
  high: number;
  low: number;
  close: number;
}

export interface Position {
  id: string;
  symbol: string;
  assetType: AssetType;
  side: OrderSide;
  quantity: number;
  averagePrice: number;
  currentPrice: number;
  marketValue: number;
  pnl: number;
  pnlPercent: number;
  delta?: number;
  gamma?: number;
  theta?: number;
  vega?: number;
  impliedVolatility?: number;
  status: "open" | "closing";
}

export interface Portfolio {
  totalValue: number;
  dailyPnl: number;
  dailyPnlPercent: number;
  totalReturn: number;
  totalReturnPercent: number;
  buyingPower: number;
  invested: number;
  openPositions: number;
  cash: number;
}

export interface Trade {
  id: string;
  symbol: string;
  assetType: AssetType;
  side: OrderSide;
  quantity: number;
  price: number;
  pnl?: number;
  status: "filled" | "pending" | "cancelled";
  timestamp: string;
}

export interface OrderRequest {
  symbol: string;
  side: OrderSide;
  quantity: number;
  orderType: OrderType;
  timeInForce: TimeInForce;
  price?: number;
}

export interface OrderResponse {
  orderId: string;
  symbol: string;
  side: OrderSide;
  quantity: number;
  orderType: OrderType;
  timeInForce: TimeInForce;
  price: number;
  status: "filled" | "pending" | "rejected";
  timestamp: string;
  message: string;
}

export interface AgentSignal {
  symbol: string;
  action: OrderSide;
  contract: string;
  strike: number;
  expiry: string;
  impliedVolatility: number;
  expectedRealizedVolatility: number;
  edge: number;
  confidence: number;
  suggestedSize: number;
  factors: string[];
}

export interface AgentRisk {
  portfolioExposure: number;
  dailyLoss: number;
  dailyLossLimit: number;
  maxPositionRisk: number;
  portfolioDelta: number;
  portfolioVega: number;
  concentration: "low" | "moderate" | "high";
  status: "safe" | "warning" | "blocked";
}

export interface DecisionEvent {
  id: string;
  timestamp: string;
  title: string;
  description: string;
  status: "completed" | "running" | "warning";
}

export interface AgentState {
  status: AgentStatus;
  regime: MarketRegime;
  confidence: number;
  riskLevel: RiskLevel;
  latestSignal: AgentSignal | null;
  risk: AgentRisk;
  timeline: DecisionEvent[];
  lastDecision: string;
}

File: backend/app/schemas/trading.py
from pydantic import BaseModel, Field
from typing import List, Optional

class MarketAsset(BaseModel):
    symbol: str
    name: str
    price: float
    change: float
    changePercent: float
    volume: int

class PortfolioSummary(BaseModel):
    totalValue: float
    dailyPnl: float
    dailyPnlPercent: float
    buyingPower: float
    openPositions: int

class PositionDTO(BaseModel):
    symbol: str
    name: str
    quantity: float
    averagePrice: float
    currentPrice: float
    pnl: float
    pnlPercent: float

class TradeDTO(BaseModel):
    id: str
    symbol: str
    action: str  # BUY, SELL
    quantity: int
    price: float
    status: str  # FILLED, REJECTED, SUBMITTED
    timestamp: str

class AgentSignal(BaseModel):
    symbol: str
    action: str  # BUY, SELL, HOLD
    confidence: float
    expectedReturn: float
    risk: str  # LOW, MEDIUM, HIGH
    reason: List[str]

class AgentDecisionDTO(BaseModel):
    id: str
    title: str
    description: str
    timestamp: str
    status: str  # completed, in_progress, pending

class AgentStateResponse(BaseModel):
    status: str  # ACTIVE, IDLE, PAUSED
    currentSymbol: str
    confidence: float
    riskScore: float
    signal: AgentSignal
    decisions: List[AgentDecisionDTO] = Field(default_factory=list)

class OrderCreateRequest(BaseModel):
    symbol: str = Field(min_length=1, description="Symbol name e.g. NVDA")
    action: str = Field(pattern="^(BUY|SELL|buy|sell)$", description="Order action BUY or SELL")
    quantity: int = Field(gt=0, description="Quantity must be greater than 0")
    orderType: str = Field(default="MARKET", description="MARKET or LIMIT")
    limitPrice: Optional[float] = Field(default=None, description="Limit price if LIMIT order")
